ocw/lib/gce.py

```diff
@@ -3,31 +3,39 @@
 import googleapiclient.discovery
 from google.oauth2 import service_account
 from dateutil.parser import parse
+from .decorators import filterService
+from googleapiclient.errors import HttpError
 import re
 
 
 class GCE(Provider):
     __instances = dict()
 
-    def __new__(cls, vault_namespace):
-        if vault_namespace not in GCE.__instances:
-            GCE.__instances[vault_namespace] = self = object.__new__(cls)
-            self.__credentials = GCECredential(vault_namespace)
-            self.__compute_client = None
-            self.__project = None
+    def __new__(cls, vault_namespace, *args, **kwargs):
+        if vault_namespace in GCE.__instances:
```

Comment:
if you will check `azure` and `ec2` you will see that they also have `__new__` and `__init__` and I can tell you that before they were the same as `GCE` class . Actually I kinda surprised that GCE still use old stile . So generally +1 to this change . But you did it a bit differently than it is done in `azure` / `ec2` . I can not find any issue with THAT approach , but I would like to discuss this with you ...

Reply:
i will take a look on the other modules. As for now i have no idea.

Reply:
ok